src/main/scala/com/cognite/sdk/scala/v1/Client.scala

```diff
@@ -25,35 +26,61 @@ final case class RequestSession[F[_]: Monad](
         .readTimeout(90.seconds)
     ).send()(sttpBackend, implicitly)
 
+  private val sttpRequest = sttp
+    .followRedirects(false)
+    .auth(auth)
+    .header("x-cdp-sdk", s"${BuildInfo.organization}-${BuildInfo.version}")
+    .header("x-cdp-app", applicationName)
+    .readTimeout(90.seconds)
+    .parseResponseIfMetadata(_.contentLength.forall(_ > 0))
+
+  private def parseResponse[T, R](uri: Uri, mapResult: T => R)(
+      implicit decoder: Decoder[Either[CdpApiError, T]]
+  ) =
+    asJson[Either[CdpApiError, T]].mapWithMetadata(
+      (response, metadata) =>
+        response match {
+          case Left(value) => throw value.error
+          case Right(Left(cdpApiError)) =>
+            throw cdpApiError.asException(uri"$uri", metadata.header("x-request-id"))
+          case Right(Right(value)) => mapResult(value)
+        }
+    )
+
+  private def unsafeBody[R](uri: Uri, response: Response[R]): R =
+    try {
+      response.unsafeBody
+    } catch {
+      // sttp's .unsafeBody returns a NoSuchElementException if the status wasn't
+      // a 200 and there is no body to return.
+      case _: NoSuchElementException =>
+        val code = response.code.toInt
+        throw SdkException(
+          s"Unexpected status code $code",
+          Some(uri),
+          response.header("x-request-id")
+        )
+      case NonFatal(_) =>
+        throw SdkException(
+          s"Unexpected exception while reading response body",
+          Some(uri),
+          response.header("x-request-id")
+        )
```

Comment:
Do we really want to swallow the NonFatal here? For example, if I get some sort of network error, I would like to know. Or is that handled elsewhere?

Reply:
Most network errors will occur before this happens, as we've received a response and we've started reading the response body at this point. However, it is possible that something goes wrong with the network as we're reading the body, of course.

I can include the exception the reply? It's not easy to see what kind of exceptions we can get that shouldn't be converted to `SdkException` here.

Reply:
*Exception message.

Reply:
I see, in that case I think it makes a lot of sense to do a catch-all here.